catalog/spatial/registry/registry-federation-admin-service-impl/src/test/java/org/codice/ddf/registry/federationadmin/service/impl/RegistryPublicationServiceImplTest.java

```diff
@@ -0,0 +1,227 @@
+        metacard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.PUBLISHED_LOCATIONS,
+                Collections.singletonList(REGISTRY_STORE_REGISTRY_ID)));
+        when(federationAdminService.getRegistryMetacardsByRegistryIds(any(List.class))).thenReturn(
+                Collections.singletonList(metacard));
+        registryPublicationService.unpublish(unPublishThisRegistryId, REGISTRY_STORE_REGISTRY_ID);
+        verify(federationAdminService).deleteRegistryEntriesByRegistryIds(Collections.singletonList(
+                unPublishThisRegistryId), Collections.singleton(REGISTRY_STORE_ID));
+        verify(federationAdminService).updateRegistryEntry(any(Metacard.class));
+    }
+
+    @Test
+    public void testUnpublishNoCurrentlyPublished() throws Exception {
+        metacard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.PUBLISHED_LOCATIONS,
+                Collections.emptyList()));
+        when(federationAdminService.getRegistryMetacardsByRegistryIds(any(List.class))).thenReturn(
+                Collections.singletonList(metacard));
+        registryPublicationService.unpublish("regId", "sourceId");
+        verify(federationAdminService,
+                never()).deleteRegistryEntriesByRegistryIds(Collections.singletonList("regId"),
+                Collections.singleton("sourceId"));
+        verify(federationAdminService, never()).updateRegistryEntry(metacard);
+    }
+
+    @Test
+    public void testUpdateNoPublications() throws Exception {
+        MetacardImpl mcard = new MetacardImpl();
+        mcard.setAttribute(RegistryObjectMetacardType.PUBLISHED_LOCATIONS, new ArrayList<>());
+        registryPublicationService.update(mcard);
+        verify(federationAdminService, never()).updateRegistryEntry(mcard);
+    }
+
+    @Test
+    public void testUpdateEmptyPublications() throws Exception {
+        MetacardImpl mcard = new MetacardImpl();
+        registryPublicationService.update(mcard);
+        verify(federationAdminService, never()).updateRegistryEntry(mcard);
+    }
+
+    @Test
+    public void testUpdateNoLastPublishDate() throws Exception {
+        MetacardImpl mcard = new MetacardImpl();
+        mcard.setAttribute(RegistryObjectMetacardType.PUBLISHED_LOCATIONS, "mylocation");
+        registryPublicationService.update(mcard);
+        verify(federationAdminService, times(1)).updateRegistryEntry(mcard);
+    }
+
+    @Test
+    public void testUpdate() throws Exception {
+        String publishedLocation = "myLocation";
+        MetacardImpl mcard = new MetacardImpl();
+        mcard.setAttribute(RegistryObjectMetacardType.PUBLISHED_LOCATIONS, publishedLocation);
+        registryPublicationService.update(mcard);
+        verify(federationAdminService, times(1)).updateRegistryEntry(mcard,
+                Collections.singleton(publishedLocation));
+        verify(federationAdminService, times(1)).updateRegistryEntry(any(Metacard.class));
```

Comment:
Is it possible to test the actual metacard update? For example, if the metacard attribute published locations has been updated to the new value.